nucliadb/src/nucliadb/common/external_index_providers/base.py

```diff
@@ -19,42 +19,131 @@
 #
 import abc
 import logging
+from enum import Enum
+from typing import Any, Iterator, Optional
 
+from pydantic import BaseModel
+
+from nucliadb_protos.nodereader_pb2 import SearchRequest
 from nucliadb_protos.noderesources_pb2 import Resource
 
 logger = logging.getLogger(__name__)
 
 
+class ExternalIndexProviderType(Enum):
+    """
+    Enum for the different external index providers.
+    For now only Pinecone is supported, but we may add more in the future.
+    """
+
+    PINECONE = "pinecone"
+
+
+class TextBlockMatch(BaseModel):
+    """
+    Model a text block/paragraph retrieved from an external index with all the information
+    needed in order to later hydrate retrieval results.
+    """
+
+    id: str
+    resource_id: str
+    field: str
+    index: int
+    position_start: int
+    position_end: int
+    split: Optional[str] = None
+    text: Optional[str] = None
+    score: float
+    order: int
```

Comment:
Could you use more verbose names? Something like
```suggestion
    id: str
    resource_id: str
    field_id: str
    index: int
    position_start: int
    position_end: int
    subfield_id: Optional[str] = None
    text: Optional[str] = None
    score: float
    order: int
```

Also, isn't `index` specific to vectors? Shouldn't be this an optional if we want to use this model for paragraphs?